alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs

```diff
@@ -0,0 +1,457 @@
+    HasField "_minUTxOValue" (Core.PParams era) Coin,
+    HasField "_maxTxSize" (Core.PParams era) Natural,
+    HasField "_prices" (Core.PParams era) Prices,
+    HasField "_maxTxExUnits" (Core.PParams era) ExUnits,
+    -- We fix Core.Tx, Core.Value, Core.TxBody, and Core.TxOut
+    Core.TxOut era ~ Alonzo.TxOut era,
+    Core.Value era ~ Alonzo.Value (Crypto era),
+    Core.TxBody era ~ Alonzo.TxBody era,
+    Core.TxOut era ~ Alonzo.TxOut era
+  ) =>
+  TransitionRule (AlonzoUTXO era)
+utxoTransition = do
+  TRC (Shelley.UtxoEnv slot pp stakepools genDelegs, u, tx) <- judgmentContext
+  let Shelley.UTxOState utxo _deposits _fees ppup = u
+
+  let txb = txbody tx
+
+  inInterval slot (getField @"vldt" txb)
+    ?! OutsideValidityIntervalUTxO (getField @"vldt" txb) slot
+
+  txins @era txb /= Set.empty ?! InputSetEmptyUTxO
+
+  -- feesOK pp tx utxo
+
+  let minimumFee = minfee pp tx
+      txFee = getField @"txfee" txb
+  minimumFee <= txFee ?! FeeTooSmallUTxO minimumFee txFee
+
+  eval (txins @era txb ⊆ dom utxo)
+    ?! BadInputsUTxO (eval ((txins @era txb) ➖ (dom utxo)))
+
+  ni <- liftSTS $ asks networkId
+  let addrsWrongNetwork =
+        filter
+          (\a -> getNetwork a /= ni)
+          (fmap (getField @"address") $ toList $ getField @"outputs" txb)
+  null addrsWrongNetwork ?! WrongNetwork ni (Set.fromList addrsWrongNetwork)
+  let wdrlsWrongNetwork =
+        filter
+          (\a -> getRwdNetwork a /= ni)
+          (Map.keys . unWdrl . getField @"wdrls" $ txb)
+  null wdrlsWrongNetwork
+    ?! WrongNetworkWithdrawal
+      ni
+      (Set.fromList wdrlsWrongNetwork)
+
+  let consumed_ = consumed pp utxo txb
+      produced_ = Shelley.produced @era pp stakepools txb
+  consumed_ == produced_ ?! ValueNotConservedUTxO consumed_ produced_
+
+  -- process Protocol Parameter Update Proposals -- NOT SURE WHAT IS GOING On HERE
+  _ppup' <-
+    trans @(Core.EraRule "PPUP" era) $
+      TRC (PPUPEnv slot pp genDelegs, ppup, txUpdates txb)
+
+  -- Check that the mint field does not try to mint ADA. This is equivalent to
+  -- the check `adaPolicy ∉ supp mint tx` in the spec.
+  Val.coin (getField @"mint" txb) == Val.zero ?! TriesToForgeADA
+
+  let outputs = Map.elems $ unUTxO (txouts @era txb)
+      minUTxOValue = getField @"_minUTxOValue" pp
+      outputsTooSmall =
+        filter
+          ( \out ->
+              let v = getField @"value" out
+               in not $
+                    Val.pointwise
+                      (>=)
+                      v
+                      (Val.inject $ scaledMinDeposit v minUTxOValue)
+          )
+          outputs
+  null outputsTooSmall ?! OutputTooSmallUTxO outputsTooSmall
+
+  let outputsTooBig =
+        filter
+          ( \out ->
+              let v = getField @"value" out
+               in (BSL.length . serialize) v > 4000
+              -- TODO this is arbitrary, THERE IS SUPPOSEDLY A NEW
+              -- PParams files that holds this what is it? --TODO fix this
```

Comment:
We have `adaPerUTxOByte`, which would seem to match what the spec wants here

Reply:
Well I know how long the script is, and I know the cost per byte, so I can compute its cost, but when is the cost too large?  What do we compare it with?

Reply:
Ok I added a function outputSize, so we now have

coin(value  out)  >= (outputSize  out <×> (adaPerUTxOByte  pp))